shares/shares.go

```diff
@@ -90,33 +72,38 @@ func (s Share) IsCompactShare() (bool, error) {
 	return isCompact, nil
 }
 
+// GetSigner returns the signer of the share, if the
+// share is not of type v1 and is not the first share in a sequence
+// it returns nil
+func GetSigner(share Share) []byte {
+	infoByte := share.InfoByte()
+	if infoByte.Version() != ShareVersionOne {
+		return nil
+	}
+	if !infoByte.IsSequenceStart() {
+		return nil
+	}
+	startIndex := namespace.NamespaceSize + ShareInfoBytes + SequenceLenBytes
+	endIndex := startIndex + blob.SignerSize
+	return share.data[startIndex:endIndex]
```

Comment:
doesn't it make sense to check whether the length of the `share.data` can contain the given range? to avoid any slice out of bounds issues